packages/ethereum/ethereum-types/src/consensus/beacon_block.rs

```diff
@@ -0,0 +1,102 @@
+//! This module defines types related to beacon's block api endpoints.
+
+use alloy_primitives::{Address, Bloom, Bytes, B256, U256};
+use serde::{Deserialize, Serialize};
+use serde_with::{serde_as, DisplayFromStr};
+
+use super::{bls::BlsSignature, sync_committee::SyncAggregate};
+
+/// A beacon block
+#[serde_as]
+#[derive(Default, Debug, Clone, Eq, PartialEq, Serialize, Deserialize)]
+pub struct BeaconBlock {
+    /// The beacon block message
+    pub message: BeaconBlockMessage,
+    /// The beacon block signature
+    pub signature: BlsSignature,
+}
+
+/// A beacon block message
+#[serde_as]
+#[derive(Default, Debug, Clone, Eq, PartialEq, Serialize, Deserialize)]
+pub struct BeaconBlockMessage {
+    /// The beacon block slot
+    #[serde_as(as = "DisplayFromStr")]
+    pub slot: u64,
+    /// The beacon block proposer index
+    #[serde_as(as = "DisplayFromStr")]
+    pub proposer_index: u64,
+    /// The beacon block parent root
+    pub parent_root: B256,
+    /// The beacon block state root
+    pub state_root: B256,
+    /// The beacon block body
+    pub body: Body,
+}
+
+/// The body of a beacon block
+#[serde_as]
+#[derive(Default, Debug, Clone, Eq, PartialEq, Serialize, Deserialize)]
+pub struct Body {
+    /// The sync aggregate of the beacon block
+    pub sync_aggregate: SyncAggregate,
+    /// The execution payload for the beacon block
+    pub execution_payload: ExecutionPayload,
+    // Values not represented in this type:
```

Comment:
I didn't bother to implement all of this, because we don't need it now, and because it's a really deep and complex structure in many of these fields